src/hardware.cpp

```diff
@@ -923,7 +928,21 @@ bool fba::Hardware::position_xy_bad(int32_t loc, fbg::dpair const & xy) const {
     // Check inner keepout region.
     auto centerxy = loc_pos_curved_mm.at(loc);
     double r = ::hypot(xy.first - centerxy.first, xy.second - centerxy.second);
-    double r_min = ::abs(loc_theta_arm.at(loc) - loc_phi_arm.at(loc));
+    //double r_min = ::abs(loc_theta_arm.at(loc) - loc_phi_arm.at(loc));
+    double r_min = (loc_theta_arm.at(loc) - loc_phi_arm.at(loc));
+    fba::Logger & logger = fba::Logger::get();
+    if (fa_behavior == -1) {
+	    logger.debug("FIBERASSIGN_BEHAVIOR not defined.  Using ::fabs correctly on r_min");
+	    r_min = ::fabs(r_min);
+    } else if (fa_behavior == 0) {
+            logger.debug("FIBERASSIGN_BEHAVIOR = 0 -- using integer ::abs on r_min for old 'buggy' behavior");
+            r_min = ::abs((int)r_min);
+    } else {
+	    logger.debug("FIBERASSIGN_BEHAVIOR = 1 -- using ::fabs correctly on r_min");
+	    r_min = ::fabs(r_min);
+    }
+    std::cout << "R_MIN = " << r_min << std::endl;
```

Comment:
note to ourselves: we ll want to disable this printing of `R_MIN`, as it "pollutes" the fiberassign-TILEID.log files.
here s an example of a file run with that: https://data.desi.lbl.gov/desi/users/sybenzvi/fa_holding_pen_bugfix_issue_258_main_20250324/002/fiberassign-002212.log